### PR TITLE
Pimp up this thing

### PR DESCRIPTION
**Pull request by msfjarvis**
Stuff :smirk:

### CODE REVIEW

**Comment by frap129** [comment]
Wow, thanks for all of this. My only question, what's the advantage of having the prop variables in a separate file?

**Comment by frap129** [comment]
Ignore that code review, I'll test when I get home and merge. Still wondering about the props file though

EDIT: Apparently only I can see the code review message

**Comment by msfjarvis** [comment]
@frap129 I've removed the separate Props class and moved the variables from it into the Utils class.

**Comment by frap129** [comment]
@MSF-Jarvis Testing everything now. Thanks again for this, my hard drive actually died while I was working on some of these things I hadn't got around to restarting them

**Comment by msfjarvis** [comment]
Rebased the PR on top of the current master

File: app/src/main/java/org/frap129/spectrum/MainActivity.java [hunk 1]
```diff
@@ -272,5 +277,26 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 return super.onOptionsItemSelected(item);
         }
     }
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults) {
+        switch (requestCode) {
```
Comment (frap129) [inline app/src/main/java/org/frap129/spectrum/MainActivity.java:282]: this switch should have a default case

**Comment by frap129** [comment]
Everything seems to work, thanks again!
Edit: Merge conflicts, will fix up when I get home

**Comment by ghost** [comment]
Yay

**Comment by msfjarvis** [comment]
Why do I even try

**Comment by frap129** [comment]
shit thought I merged it, forgot about this. Sorry man, I'll actually merge it
 this time

**Pull request merged**